app/src/features/Preview/Preview.tsx

```diff
@@ -0,0 +1,262 @@
+  },
+  texts: {
+    textAlign: "center",
+  },
+  cellsTitle: {
+    background: theme.palette.background.paper,
+  },
+  rowBorder: {
+    borderBottom: `1px solid ${theme.palette.divider}`,
+  },
+  preview: {
+    padding: 0,
+    width: "100%",
+    borderRadius: theme.shape.borderRadius,
+    border: `1px solid ${theme.palette.divider}`,
+    "& > div": {
+      padding: theme.spacing(3),
+      borderRadius: theme.shape.borderRadius,
+    },
+  },
+  button: {
+    alignSelf: "flex-start",
+    marginBottom: theme.spacing(2),
+    textTransform: "none",
+    color: theme.palette.text.secondary,
+    "&:hover": {
+      backgroundColor: "inherit",
+      color: theme.palette.text.primary,
+    },
+  },
+}));
+
+type Exploration = { fields: string[]; rows: string[][] };
+
+const Preview = (): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const history = useHistory();
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+  const { mappingId } = useParams<{
+    mappingId?: string;
+  }>();
+
+  const [alert, setAlert] = useState<string | undefined>(undefined);
+  const handleAlertClose = () => setAlert(undefined);
+
+  const [preview, setPreview] = useState<Record<string, unknown> | undefined>(
+    undefined
+  );
+
+  const { data: mapping } = useApiResourcesRetrieveQuery(
+    { id: mappingId ?? "" },
+    { skip: mappingId === undefined }
+  );
+
+  const { data: owner } = useApiOwnersRetrieveQuery(
+    {
+      id: mapping?.primary_key_owner ?? "",
+    },
+    { skip: mapping === undefined }
+  );
+
+  const {
+    data: exploration,
+    isLoading: isExplorationLoading,
+  } = usePagaiExploreRetrieveQuery(
+    {
+      resourceId: mapping?.id ?? "",
+      owner: owner?.name ?? "",
+      table: mapping?.primary_key_table ?? "",
+    },
+    { skip: mapping === undefined || owner === undefined }
+  );
+
+  const [apiPreviewCreate] = useApiPreviewCreateMutation();
+
+  const handleFhirIconClick = (index: number) => async () => {
+    const primaryKey = mapping?.primary_key_table;
+    if (exploration && primaryKey && mapping) {
+      const indexPrimaryKey = (exploration as Exploration).fields.indexOf(
+        primaryKey
+      );
+      const primaryKeyValue = (exploration as Exploration).rows[index]?.[
+        indexPrimaryKey
+      ];
+      try {
+        const previewResult = await apiPreviewCreate({
+          previewRequest: {
+            resource_id: mapping?.id,
+            primary_key_values: [primaryKeyValue ?? ""],
+          },
+        });
+        setPreview(previewResult);
+      } catch (e) {
+        setAlert(e.message);
+      }
+    }
+  };
+
+  const handleCancelClick = () => {
+    history.goBack();
+  };
+
+  const FhirIconCell = () => (
+    <IconButton size="small" className={classes.icon}>
+      <Icon
+        icon={IconNames.FLAME}
+        iconSize={15}
+        className={classes.iconFlame}
+      />
+    </IconButton>
+  );
```

Comment:
I don't really see the point of creating a component from within another component, even more when this is only used once

Reply:
Moreover, this component will be recomputed on each render, this doesn't really bring any improvement in my opinion